lua/nvim-treesitter-textobjects/shared.lua

```diff
@@ -0,0 +1,754 @@
+  return fn(range, query_string, query_group, bufnr)
+end
+
+---@param range TSTextObjects.Range
+---@param query_string string
+---@param query_group string
+---@param bufnr integer
+---@return TSTextObjects.Range?
+function M.next_textobject(range, query_string, query_group, bufnr)
+  local node_end = range.end_byte
+  local search_start = node_end
+
+  ---@param match TSTextObjects.Match
+  ---@return boolean
+  local function filter_function(match)
+    if match.range == range then
+      return false
+    end
+    if range.parent_id == match.range.parent_id then
+      local start = match.range.start_byte
+      local end_ = match.range.end_byte
+      return start >= search_start and end_ >= node_end
+    end
+    return false
+  end
+
+  ---@param match TSTextObjects.Match
+  ---@return integer
+  local function scoring_function(match)
+    local node_start = match.range.start_byte
+    return -node_start
+  end
+
+  local next_range = M.find_best_match(bufnr, query_string, query_group, filter_function, scoring_function)
+
+  return next_range and next_range.range
+end
+
+---@param range TSTextObjects.Range
+---@param query_string string
+---@param query_group? string
+---@param bufnr integer
+---@return TSTextObjects.Range?
+function M.previous_textobject(range, query_string, query_group, bufnr)
+  query_group = query_group or "textobjects"
+
+  local node_start = range.start_byte
+  local search_end = node_start
+
+  ---@param match TSTextObjects.Match
+  ---@return boolean
+  local function filter_function(match)
+    if range.parent_id == match.range.parent_id then
+      local end_ = match.range.end_byte
+      local start = match.range.start_byte
+      return end_ <= search_end and start < node_start
+    end
+    return false
+  end
+
+  ---@param match TSTextObjects.Match
+  ---@return integer
+  local function scoring_function(match)
+    local node_end = match.range.end_byte
+    return node_end
+  end
+
+  local previous_node = M.find_best_match(bufnr, query_string, query_group, filter_function, scoring_function)
+
+  return previous_node and previous_node.range
+end
+
+---@param bufnr integer
+---@param query_group? string
+---@param queries? string[]
+--- @return boolean
+function M.check_support(bufnr, query_group, queries)
+  query_group = query_group or "textobjects"
+
+  local buf_lang = ts.language.get_lang(vim.bo[bufnr].filetype)
+  if not buf_lang then
+    return false
+  end
+  local parser = ts.get_parser(bufnr, buf_lang)
+  if not parser then
+    return false
+  end
+  -- TODO (TheLeoP): should be cached?
+  local ok, _queries = pcall(ts.query.get, buf_lang, query_group)
+  if not ok or not _queries then
+    return false
+  end
+
+  if queries then
+    if vim.tbl_isempty(queries) then
+      return false
+    end
+
+    local available_textobjects = M.available_textobjects(buf_lang, query_group)
+    for _, query in ipairs(queries) do
+      if not vim.list_contains(available_textobjects, query:sub(2)) then
```

Comment:
```suggestion
      if not vim.tbl_contains(available_textobjects, query:sub(2)) then
```

I believe this is the only change necessary to maintain support for non-neovim nightly